test/org/javarosa/xform/parse/XFormParserTest.java

```diff
@@ -300,6 +301,16 @@ public void serializeAndRestoreMetaNamespaceFormInstance() throws IOException {
         assertEquals("/data/text", submissionProfile.getRef().getReference().toString());
     }
 
+    /**
+     * Simple tests that documents assumption that the model has to come before the body tag.
+     * According to the comment above {@link XFormParser#parseModel(Element)} method,
+     * this is not mandated by the specs but has been implemented this way to keep parsing simpler.
+     */
```

Comment:
It’s good to have this implementation requirement documented, but I don’t see how this test helps. If we change the implementation to remove the restriction this test doesn’t fail. If someone in the field makes a form that violates the restriction, this test won’t catch it.

Reply:
@dcbriccetti Not sure if I follow. Once the restriction is removed, it should start failing because it expects a `RuntimeException` to be thrown during parsing by this code
`@Test(expected = RuntimeException.class)`.